mapper/ui/location_marker.lua

```diff
@@ -0,0 +1,40 @@
+amap.ui['location_marker'] = amap.ui['location_marker'] or {
+    location_ids_marked = {},
+    working = false,
+    color = 'pale_violet_red',
+    color2 = 'alice_blue',
+    event_handlers = {}
+}
+
+function amap.ui.location_marker:toggle_location_marker()
+    if amap.ui.location_marker.working then
+        for _, event_id, _ in pairs(amap.ui.location_marker['event_handlers']) do
+            scripts.event_register:kill_event_handler(event_id)
+        end
+        amap.ui.location_marker['event_handlers'] = {}
+        for location_id, _ in pairs(amap.ui.location_marker.location_ids_marked) do
+            unHighlightRoom(location_id)
+        end
+        amap.ui.location_marker['location_ids_marked'] = {}
+        amap.ui.location_marker.working = false
+        amap:print_log('wylaczylem markowanie lokacji')
+    else
+        table.insert(amap.ui.location_marker['event_handlers'], scripts.event_register:register_event_handler('amapNewLocation', [[ amap.ui.location_marker.mark_current_new_location ]]))
+        table.insert(amap.ui.location_marker['event_handlers'], scripts.event_register:register_event_handler('amapBlockerFired', [[ amap.ui.location_marker.blocked_by_blocker ]]))
+        amap.ui.location_marker.working = true
+        amap:print_log('wlaczylem markowanie lokacji')
+    end
+
+end
+
+function amap.ui.location_marker:mark_current_new_location(...)
+    local new_location_id = arg[1]
+    local rgb_color = color_table[amap.ui.location_marker.color]
+    local rgb_color2 = color_table[amap.ui.location_marker.color2]
+    highlightRoom(new_location_id, rgb_color[1], rgb_color[2], rgb_color[3], rgb_color2[1], rgb_color2[2], rgb_color2[3], 0.85, 255, 255)
```

Comment:
A nie myślałeś o tym, żeby skorzystać z tego Highlightera co tam wcześniej napisałem?

Reply:
Tzn. jak coś to merguj :)

Reply:
Lol, faktycznie! :D To bylo celem testow i juz nie zdazylem przerzucic na Twojego highlightera. Zaraz zmienie